lib/index.js

```diff
@@ -0,0 +1,240 @@
+    // catastrophic case where remaining `2f` electors cannot communicate
+    // (recovery electors are not a failsafe against a coordinated byzantine
+    // attack as the byzantine electors could collude to show different sides
+    // of a fork, but safe against network failures preventing consensus)
+    recoveryElectors,
+    // number of generations of merge events where no new electors participate
+    // and fewer than 2f+1 total electors (including self) participate causing
+    // recovery mode to trigger because consensus cannot be reached
+    recoveryGenerationThreshold,
+  };
+};
+
+// it is useful to override this function in tests
+api._computeElectors = async ({
+  blockHeight, electorPoolElectors, latestBlockSummary, ledgerConfiguration,
+  ledgerNode
+}) => {
+  // get partipicants for the last block
+
+  // FIXME: if we're using the _blocks API here, should it be made public
+  // (remove the underscore?)
+
+  const {_blocks} = ledgerNode.consensus;
+  const {consensusProofPeers, mergeEventPeers} = await _blocks.getParticipants(
+    {blockHeight: blockHeight - 1, ledgerNode});
+
+  // TODO: we should be able to easily remove previously detected
+  // byzantine nodes (e.g. those that forked at least) from the electors
+
+  // TODO: simply count consensus event signers once and proof signers
+  //   twice for now -- add comprehensive elector selection and
+  //   recommended elector vote aggregating algorithm in v2
+  const aggregate = {};
+  // add merge event peers
+  for(const id of mergeEventPeers) {
+    aggregate[id] = {id, weight: 1};
+  }
+
+  // TODO: weight previous electors more heavily to encourage continuity
+  // add consensusProof peers
+  for(const id of consensusProofPeers) {
+    if(id in aggregate) {
+      aggregate[id].weight = 3;
+    } else {
+      aggregate[id] = {id, weight: 2};
+    }
+  }
+  // add peers from the electorPool
+  for(const elector of electorPoolElectors) {
+    const electorType = [].concat(elector.type);
+    // electors defined in *any* way in the elector pool document should
+    // have the greatest weight
+    // FIXME: weight these like this?
+    // FIXME: is type === Continuity2017Peer a real thing?  If so, what do we
+    // do for that.
+    let weight = 100;
+    let recoveryElector = false;
+    if(electorType.includes(ELECTOR_TYPE_RECOVERY)) {
+      weight += 100;
+      recoveryElector = true;
+    }
+    if(electorType.includes(ELECTOR_TYPE_GUARANTOR)) {
+      weight += 100;
+    }
+    aggregate[elector.id] = {id: elector.id, recoveryElector, weight};
+  }
+
+  let electors = Object.values(aggregate);
+
+  let electorCount;
+  // the electorPool document may not be defined yet
+  if(electorPoolElectors.length > 0) {
+    // FIXME: `electorCount` in the ledgerConfiguration is used to specify a
+    // *maximum* number of electors correct?
+    // Therefore, electorCount is a maxiumum, so why would
+    // `electorPoolElectors.length` be considered a maximum?
```

Comment:
I don't understand the question ... this is doing `Math.min` here. We can't select more electors than are in the pool, so it's a limit. However, we also can't select more than `electorCount` so it's another limit. The smallest of these two is what we must choose, i.e. code seems to make sense to me.

Reply:
> We can't select more electors than are in the pool,

That's the part I don't understand.  The electorPool contains guarantors and recovery electors, and then we're adding to that number by pulling electors out of merge events/consensus.  I must be missing something.

Reply:
Oh, yes, you're missing something! We are absolutely NOT pulling electors out of merge events and consensus. We're ONLY pulling from the pool!

Reply:
If you aren't in the pool, you don't get a spot. If you're in the pool but you're not participating, you don't get a spot.

If you're in the pool and you're an active participant, you've got a random chance of getting picked.

Reply:
mind blown...  I now know with certainty that I have no idea how this is supposed to work.